flake.nix

```diff
@@ -0,0 +1,47 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-22.11";
+    systems.url = "github:nix-systems/default";
```

Comment:
Let's remove this and either go with flake-utils or snowfall-lib. Flake utils would work fine for now and looks similar to your existing implementation: https://github.com/numtide/flake-utils